tests/ui/rfcs/rfc-0000-never_patterns/ICE-133063-never-arm-no-otherwise-block.rs

```diff
@@ -0,0 +1,21 @@
+#![feature(never_patterns)]
+#![feature(if_let_guard)]
+#![allow(incomplete_features)]
+
+fn split_last(_: &()) -> Option<(&i32, &i32)> {
+    None
+}
+
+fn assign_twice() {
+    loop {
+        match () {
+            (!| //~ ERROR: mismatched types
+            !) if let _ = split_last(&()) => {} //~ ERROR a never pattern is always unreachable
+            //~^ ERROR: mismatched types
+            //~^^ WARNING: irrefutable `if let` guard pattern [irrefutable_let_patterns]
+            _ => {}
+        }
+    }
+}
```

Comment:
Could you replace this with a type-correct test, so this test keeps being useful in the future if the implementation changes? In particular we may skip MIR lowering in case of a type error, which would render the current test useless.
```rust
fn foo(x: !) {
	match x {
		(! | !) if false => {}
		_ => {}
	}
}
```

Reply:
Addressed in https://github.com/rust-lang/rust/pull/135409/commits/d8a216b866ff238589005fb1ea4c54644d8d24bf.

Note: `x: !` can't avoid mismatched types error. So I use enum Void {} instead of it (other test codes in the folder do so).